org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.xtend

```diff
@@ -169,6 +175,22 @@ import static org.eclipse.xtext.diagnostics.Severity.*
 					commands = this.commandRegistry.getCommands()
 				]
 			}
+
+			val semanticHighlighting = params?.capabilities?.textDocument?.semanticHighlightingCapabilities?.semanticHighlighting ?: false;
```

Comment:
params is already dereferenced unconditionally above. No need for `?.`.
Also: should extract capabilities into a local var and ruse that one.